File: src/main/java/com/j148/backend/contractor/model/Contractor.java
package com.j148.backend.contractor.model;
import com.j148.backend.contract_period.model.ContractPeriod;
import com.j148.backend.user.model.User;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Contractor {
    @Min(value = 1, message = "Contractor ID cannot be less than 1")
    private Long contractorId;
   
    private Status status;
    private User user;
    private ContractPeriod contractPeriod;

    public enum Status {
        EXTERNAL, SUSPENDED, ACTIVE, ON_LEAVE
    }
}

File: hrmsPython/models/Contractor.py
from dataclasses import dataclass
from typing import Optional
from enum import Enum

from dataclasses_json import dataclass_json

from models import User, ContractPeriod


class Status(Enum):
    EXTERNAL = "EXTERNAL"
    SUSPENDED = "SUSPENDED"
    ACTIVE = "ACTIVE"
    ON_LEAVE = "ON_LEAVE"

@dataclass_json
@dataclass
class Contractor:
    contractorId: Optional[int] = None  # Maps to Long in Java
    status: Optional[Status] = None     # Maps to Status enum in Java
    user: Optional[User] = None         # Maps to User class in Java
    contractPeriod: Optional[ContractPeriod] = None  # Maps to ContractPeriod class in Java
